internal/pipeline/tar_assembler.go

```diff
@@ -6,16 +6,18 @@ package pipeline
 // compression type, and stores the output with the given filename.
 type TarAssemblerOptions struct {
 	Filename    string `json:"filename"`
+	Size        uint64 `json:"size"`
 	Compression string `json:"compression,omitempty"`
 }
 
 func (TarAssemblerOptions) isAssemblerOptions() {}
 
 // NewTarAssemblerOptions creates a new TarAssemblerOptions object, with the
 // mandatory options set.
-func NewTarAssemblerOptions(filename string) *TarAssemblerOptions {
+func NewTarAssemblerOptions(filename string, size uint64) *TarAssemblerOptions {
```

Comment:
Hm, this won't compile, will it? You need to update the callers at the same time (ideally without any functional change).